### DISCUSSION TITLE
Login failed url with domain

### DISCUSSION THREAD

**Discussion by jssgarcia** [datalust/seq-tickets#2254] [Help]
I have individual license. Seq is installed on individual server.  User is not on Active Directory or OpenID.  Only one user.  User is Administrator.
For internal access, login successfully from both localhost and internal IP.
For access from the outside, NAT is applied, with redirection over the same port 5341.

If a try log-in from url http:/localhost;5341/ log-in successfully.
If a try log-in from external url with specific domain, the login is correct, but the login screen always appears again and again.

Steps:

1. First login http://domain-url.com:5341/#/login (login succesfully)
2. Second try (now url changed http://domain-url.com:5341/#/login?to=%2Flogin%3Fto%3D%252Flogin
3. Third try (now ulr chaned automated http://domain-url.com:5341/#/login?to=%2Flogin%3Fto%3D%252Flogin%253Fto%253D%25252Flogin%25253Fto%25253D%2525252Flogin
4. and so on.

Inside the log, all is ok.

```log
{"@t":"2024-08-20T12:21:39.8860777Z","@mt":"User {Username} logged in successfully","Username":"ambulancia","Principal":{"Id":"principal-public","OnBehalfOfUserId":null},"ActionId":"9204c6c6-d562-4d7d-b9f9-73cca654f24f","ActionName":"Seq.Server.Web.Api.UsersController.Login (Seq)","RequestId":"800002e3-000a-f300-b63f-84710c7967bb","RequestPath":"/api/users/login"}
{"@t":"2024-08-20T12:21:39.8888827Z","@mt":"Logged in a {UserId} session","UserId":"user-admin","SourceContext":"Seq.Server.Web.Api.UsersController","Principal":{"Id":"principal-public","OnBehalfOfUserId":null},"ActionId":"9204c6c6-d562-4d7d-b9f9-73cca654f24f","ActionName":"Seq.Server.Web.Api.UsersController.Login (Seq)","RequestId":"800002e3-000a-f300-b63f-84710c7967bb","RequestPath":"/api/users/login"}
{"@t":"2024-08-20T12:21:39.9415533Z","@mt":"HTTP {RequestMethod} {RequestPath} responded {StatusCode} in {Elapsed:0.0000} ms","@r":["0.5164"],"RequestMethod":"GET","RequestPath":"/api/licenses/license-server","StatusCode":401,"Elapsed":0.5164,"SourceContext":"Seq.Server.Web.Middleware.RequestCompletionMiddleware","RequestId":"800002e5-000a-f300-b63f-84710c7967bb"}
{"@t":"2024-08-20T12:21:39.9771999Z","@mt":"HTTP {RequestMethod} {RequestPath} responded {StatusCode} in {Elapsed:0.0000} ms","@r":["0.3961"],"RequestMethod":"GET","RequestPath":"/api/roles","StatusCode":401,"Elapsed":0.3961,"SourceContext":"Seq.Server.Web.Middleware.RequestCompletionMiddleware","RequestId":"8000072a-0008-f900-b63f-84710c7967bb"}
{"@t":"2024-08-20T12:22:26.4838037Z","@mt":"User {Username} logged in successfully","Username":"ambulancia","Principal":{"Id":"principal-public","OnBehalfOfUserId":null},"ActionId":"9204c6c6-d562-4d7d-b9f9-73cca654f24f","ActionName":"Seq.Server.Web.Api.UsersController.Login (Seq)","RequestId":"8000027d-0002-f000-b63f-84710c7967bb","RequestPath":"/api/users/login"}
{"@t":"2024-08-20T12:22:26.4859433Z","@mt":"Logged in a {UserId} session","UserId":"user-admin","SourceContext":"Seq.Server.Web.Api.UsersController","Principal":{"Id":"principal-public","OnBehalfOfUserId":null},"ActionId":"9204c6c6-d562-4d7d-b9f9-73cca654f24f","ActionName":"Seq.Server.Web.Api.UsersController.Login (Seq)","RequestId":"8000027d-0002-f000-b63f-84710c7967bb","RequestPath":"/api/users/login"}
{"@t":"2024-08-20T12:22:26.5200355Z","@mt":"HTTP {RequestMethod} {RequestPath} responded {StatusCode} in {Elapsed:0.0000} ms","@r":["0.7316"],"RequestMethod":"GET","RequestPath":"/api/licenses/license-server","StatusCode":401,"Elapsed":0.7316,"SourceContext":"Seq.Server.Web.Middleware.RequestCompletionMiddleware","RequestId":"8000027e-0002-f000-b63f-84710c7967bb"}
{"@t":"2024-08-20T12:22:26.5435839Z","@mt":"HTTP {RequestMethod} {RequestPath} responded {StatusCode} in {Elapsed:0.0000} ms","@r":["0.2937"],"RequestMethod":"GET","RequestPath":"/api/roles","StatusCode":401,"Elapsed":0.2937,"SourceContext":"Seq.Server.Web.Middleware.RequestCompletionMiddleware","RequestId":"8000042e-0008-ef00-b63f-84710c7967bb"}
```

but browser console log:

```log
http://domain-url:5341/api/licenses/license-server (401 Unauthorized)
http://domain-url:5341/api/roles (401 Unauthorized)
```
and each try same info on cosole log.

Any solution?

Thanks.

**Comment by KodrAus** [comment]
Hi @jssgarcia :wave: If after you login the first time you remove the path and query from your URI, so it's just `http://domain-url.com:5341` do you see the events page? Or does that also redirect back to the login page?

As an aside, if you're accessing Seq externally, we'd definitely recommend configuring HTTPS (see [here for Docker-based deployments](https://docs.datalust.co/docs/https-tlsssl) and [here for Windows-based deployments](https://docs.datalust.co/docs/https-on-windows)).

**Reply by jssgarcia** [reply]
> If after you login the first time you remove the path and query from your URI, so it's just http://domain-url.com:5341 do you see the events page? Or does that also redirect back to the login page?

In this case also redirect back to the login page, again and again. 

We are also going to configure **https**, although I think it is not related to this problem.

**Reply by KodrAus** [reply]
Hmm, it sounds like the problem is somewhere between Seq and your browser on the other side of your reverse-proxy/load balancer. Some things to check are:

1. Whether you're getting a valid session cookie. The network tab in your browser should give you an idea of what's coming back from the login endpoint.
2. Whether the clock on your Seq server may be skewed, causing the login cookie to be immediately expired by the client.

It shouldn't be related here, but you can also look at setting [Seq's canonical URI](https://docs.datalust.co/docs/json-config#apicanonicaluri) to your publicly accessible endpoint. This makes sure any URIs Seq generates use the public address and not whatever it happens to be listening on locally.

**Comment by leopowershield** [comment]
I am facing the same issue. I'm using http, I'll migrate to https and we will see if it is still happening.

**Comment by leopowershield** [comment]
Here is some feedback: changing to HTTPS made the trick!

**Reply by nblumhardt** [reply]
Great! Thanks for the follow-up. It sounds like the browser is refusing to send the session cookie over an insecure connection - unsure how that would come about in these circumstances but overall fairly easy to do.
